crates/next-core/js/src/entry/app-renderer.tsx

```diff
@@ -137,26 +137,51 @@ async function runOperation(renderData: RenderData) {
     tree = [info.segment, { children: tree }, components];
   }
 
+  const proxyMethodsForModule = (
+    id: string
+  ): ProxyHandler<FlightManifest["__ssr_module_mapping__"][""]> => ({
+    get(_target, name) {
+      return {
+        id,
+        chunks: JSON.parse(id)[1],
+        name,
+      };
+    },
+  });
+  const proxyMethodsNested = (): ProxyHandler<
+    FlightManifest["__ssr_module_mapping__"]
+  > => {
+    return {
+      get(target, name, receiver) {
+        if (name === "__ssr_module_mapping__") {
+          return manifest;
+        }
+        if (name === "__entry_css_files__") {
+          return __entry_css_files__;
+        }
+        return new Proxy({}, proxyMethodsForModule(name as string));
+      },
+    };
+  };
```

Comment:
It would be helpful to document these workarounds. From the outside they are very obscure and the data flow is hard to understand, and easy to break.